codegen/src/main/java/software/amazon/awssdk/codegen/docs/PaginationDocs.java

```diff
@@ -140,14 +224,35 @@ private ClassName syncResponsePageType() {
     /**
      * @return A Poet {@link ClassName} for the return type of sync paginated operation.
      */
-    public ClassName syncPaginatedResponseType() {
+    private ClassName syncPaginatedResponseType() {
         return poetExtensions.getResponseClassForPaginatedSyncOperation(operationModel.getOperationName());
     }
 
+    /**
+     * @return A Poet {@link ClassName} for the return type of Async paginated operation.
+     */
+    private ClassName asyncPaginatedResponseType() {
+        return poetExtensions.getResponseClassForPaginatedAsyncOperation(operationModel.getOperationName());
+    }
+
     private CodeBlock noteAboutSyncNonPaginatedMethod() {
         return CodeBlock.builder()
                         .add("\n<p><b>Note: If you prefer to have control on service calls, use the {@link #$L($T)} operation."
                              + "</b></p>", operationModel.getMethodName(), requestType())
                         .build();
     }
+
+    /**
+     * @return A Poet {@link ClassName} for the reactive streams {@link Publisher}.
+     */
+    private ClassName getPublisherType() {
```

Comment:
Do we really need this and the one below in separate methods?

Reply:
These are used 2-3 places in the class. I thought it would be better to have them in separate method rather than calling the ClassName.get() method.